File: facade/facade.js
class TicketingSystem {
    constructor(movieName) {
        this.movieName = movieName;
    }

    buyTicket() {
        console.log(`Ticket purchased for ${this.movieName}`);
    }
}

class SeatReservationSystem {
    constructor(theaterName) {
        this.theaterName = theaterName;
    }

    selectSeat() {
        console.log(`Seat selected at ${this.theaterName}`);
    }
}

class SnackCounter {
    constructor(snacks) {
        this.snacks = snacks;
    }

    buySnacks() {
        console.log(`${this.snacks} purchased at the snack counter`);
    }
}

class MovieTheaterFacade {
    constructor(movieName, theaterName, snacks) {
        this.ticketingSystem = new TicketingSystem(movieName);
        this.seatReservationSystem = new SeatReservationSystem(theaterName);
        this.snackCounter = new SnackCounter(snacks);
    }

    watchMovie() {
        this.ticketingSystem.buyTicket();
        this.seatReservationSystem.selectSeat();
        this.snackCounter.buySnacks();

        console.log('Enjoy the movie!');
    }
}

const movieTheaterFacade = new MovieTheaterFacade('Movie-1', 'Theater-1', 'Burger and Coke');
movieTheaterFacade.watchMovie();

File: facade/facade.py
class TicketingSystem:
    def __init__(self, movie_name):
        self.movie_name = movie_name

    def buy_ticket(self):
        print(f'Ticket purchased for {self.movie_name}')


class SeatReservationSystem:
    def __init__(self, theater_name):
        self.theater_name = theater_name

    def select_seat(self):
        print(f'Seat selected at {self.theater_name}')


class SnackCounter:
    def __init__(self, snacks):
        self.snacks = snacks

    def buy_snacks(self):
        print(f'{self.snacks} purchased at the snack counter')


class MovieTheaterFacade:
    def __init__(self, movie_name, theater_name, snacks):
        self.ticketing_system = TicketingSystem(movie_name)
        self.seat_reservation_system = SeatReservationSystem(theater_name)
        self.snack_counter = SnackCounter(snacks)

    def watch_movie(self):
        self.ticketing_system.buy_ticket()
        self.seat_reservation_system.select_seat()
        self.snack_counter.buy_snacks()

        print('Enjoy the movie!')


if __name__ == '__main__':
    movie_theater_facade = MovieTheaterFacade('Movie-1', 'Theater-1', 'Burger and Coke')
    movie_theater_facade.watch_movie()
